### ISSUE TITLE
util.Compress is broken or produces a malformed/mistyped string

### ISSUE THREAD

**Issue by TiberiumFusion** [Facepunch/garrysmod-issues#4469]
### Details
`util.Compress()` produces garbage when interpreted as a string.

### Steps to reproduce
```
local test1 = "{\"Changes\":[{\"LAT\":17082.6186,\"UID\":\"cmw_testattachment\",\"A\":1.0}],\"Incremental\":true}"
print("test1", test1, util.Compress(test1))
-- prints: test1	{"Changes":[{"LAT":17082.6186,"UID":"cmw_testattachment","A":1.0}],"Incremental":true}	]

	
local test2 = "Hello world!"
print("test2", test2, util.Compress(test2))
-- prints: test2	Hello world!	]
```

As you can see, util.Compress produces a single `]` bracket. According to the [wiki page](https://wiki.facepunch.com/gmod/util.Compress), it will return a "compressed" string. While I don't know why string was selected as the container for the compressed bytes, the wiki labels the return type of util.Compress as a string, so I'd expect it to be able to interpret it as a string and manipulate it as a string. The `]` print result, however, indicates that the object contains either meaningless data or has lost its string typeness.

In other words, if the result from util.Compress is indeed a string, it should print much more pseudo-gibberish than a single `]`.

### Environment
Dev 2020.04.23 on Win x64

**Comment by bigdogmat** [comment]
It contains binary data which print has trouble displaying. If I had to guess it stops at a null character.

**Comment by meepen** [comment]
It doesn't, it produces null bytes which are the end of c strings, making the source console stop printing. If you want to see the full data print it in hex, otherwise you can see str:len() for the length

**Comment by TiberiumFusion** [comment]
> It doesn't, it produces null bytes which are the end of c strings, making the source console stop printing. If you want to see the full data print it in hex, otherwise you can see str:len() for the length

Yep... that totally slipped my mind. The string object isn't a full Lua string but a wrapper to some C++ side bytes. Thanks for saving my skin.

**Comment by bigdogmat** [comment]
It's still just a Lua string, they can contain binary data just fine.
